src/golang.org/exercises/sub-product-and-sum/sub-product-and-sum.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"fmt"
+)
+
+func subtractProductAndSum(n int) int {
+	if n <= 0 {
+		fmt.Printf("Result: 0 \n")
+		return 0
+	}
+
+	sum := 0
+	product := 1
+
+	for n > 0 {
+		mod := n % 10
+		sum += mod
+		product *= mod
+		n /= 10
+	}
+	result := product - sum
+	fmt.Printf("Result: %d \n", result)
```

Comment:
See my remark about i/o in computational functions here: https://github.com/kcarrel/Golang-Training-Notes/pull/1#discussion_r405216965

Reply:
moved printing to caller function in commit https://github.com/kcarrel/Golang-Training-Notes/pull/2/commits/8bb9890441e7775d259e4cc5597604ad11d1e670